### DISCUSSION TITLE
Is it possible for user to extract private key (specifically Polygon) without fully implementing the feature?

### DISCUSSION THREAD

**Discussion by aaleksandar** [Web3Auth/Web3Auth#935] [Help]
We're yet to build a full feature for our users to extract private keys from their Web3Auth-created Polygon wallet.

In the meantime, we'd like to support users to do this manually if possible.
Is there anything they can do on their end to extract their private keys without waiting on the feature implementation?

**Comment by Saviour1001** [comment]
For your particular client id and the logins you provide, the `getPrivateKey` function will need to be called by the user's end as shown [here](https://web3auth.io/docs/sdk/web/providers/#exporting-web3auth-private-key) . 

I don't think a good UX can be designed to allow users to extract the keys on their own.

**Reply by jonahcui** [reply]
I have read Self-host and mpc's documents; I found that I can get the user's private key like plug and play. Is it safe for users? Which mode should I select if I don't want to get the user's private key?
If our Dapp can get the user's private key, it may let the user untrust us.

**Reply by Saviour1001** [reply]
We have a new version of MPC SDK coming up, in which the private key doesn't reconstruct.

**Reply by jonahcui** [reply]
If the new mpc SDK API is the same as the old SDK, I will try the old API. If not, I will wait for the latest SDK. Would you like to share the new SDK roadmap with me?
